### PR TITLE
tools: perf: introduce the 'better' hint

### PR DESCRIPTION
**Pull request by janekmi**
This change is [Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)

### CODE REVIEW

**Review by lukaszstolarczuk — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi)

___
*[tools/perf/lib/Figure.py, line 158 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDe5Kz5TOg_cyNzLrj:-MhDe5Kz5TOg_cyNzLrk:bix307p) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L158)):*
> ```Python
>         output = label_by_column.get(column, column)
>         if with_better:
>             better = better_by_column.get(column, column)
> ```

soo... if you e.g. have column/metric `bw_max` - you'll get
```
bw_max
bw_max
```
shouldn't this be an empty string by default?
___
*[tools/perf/lib/Figure.py, line 159 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDeQhR3SOrYfoqUjLV:-MhDeQhR3SOrYfoqUjLW:b-cbdoow) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L159)):*
> ```Python
>         if with_better:
>             better = better_by_column.get(column, column)
>             output = better + '\n' + output
> ```

I'd personally reverse the order, firstly the column/metric and in the second row, e.g. `(higher is better)`

e.g. on our charts:

**Review by osalyk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: all files reviewed, 2 unresolved discussions (waiting on @janekmi and @lukaszstolarczuk)

___
*[tools/perf/lib/Figure.py, line 159 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDeQhR3SOrYfoqUjLV:-MhI64e34htqSel1ApIE:b-8fx2u7) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L159)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

I'd personally reverse the order, firstly the column/metric and in the second row, e.g. `(higher is better)`

e.g. on our charts:

</blockquote></details>

:+1:

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: all files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[tools/perf/lib/Figure.py, line 158 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDe5Kz5TOg_cyNzLrj:-MhI9rpV3JlBXBIQmi56:b-aji8fz) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L158)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

soo... if you e.g. have column/metric `bw_max` - you'll get
```
bw_max
bw_max
```
shouldn't this be an empty string by default?
</blockquote></details>

No. Because it may give you an impression everything is fine whereas one of them may be missing. But when you see a text not in plain English with underscores instead of spaces you win two times:
1. you know something went wrong
2. you know what the argument was which has caused an issue
___
*[tools/perf/lib/Figure.py, line 159 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDeQhR3SOrYfoqUjLV:-MhIAm9W99usoxgTEmly:b-ourtyq) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L159)):*
<details><summary><i>Previously, osalyk (Oksana Sałyk) wrote…</i></summary><blockquote>

:+1: 
</blockquote></details>

**It is obviously subjective**. I personally do not need these hints. It is obvious **to me** that e.g. having a higher bandwidth is better. So, more important **to me** is to read the unit instead of a generalized message suggesting I do not need to know what the unit is as long as I know that 'higher is better'. xD

If after my explanation both of you still think it is better to have it the other way around I will surrender.

**Review by osalyk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: all files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[tools/perf/lib/Figure.py, line 159 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDeQhR3SOrYfoqUjLV:-MhIVngCCdLJpwHI3-WJ:b-xkmio4) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L159)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

**It is obviously subjective**. I personally do not need these hints. It is obvious **to me** that e.g. having a higher bandwidth is better. So, more important **to me** is to read the unit instead of a generalized message suggesting I do not need to know what the unit is as long as I know that 'higher is better'. xD

If after my explanation both of you still think it is better to have it the other way around I will surrender.
</blockquote></details>

To me, Luke's proposal is more readable. We read from left to right.

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[tools/perf/lib/Figure.py, line 159 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDeQhR3SOrYfoqUjLV:-MhIpx1O0az1p_60UANg:b-896fix) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L159)):*
<details><summary><i>Previously, osalyk (Oksana Sałyk) wrote…</i></summary><blockquote>

To me, Luke's proposal is more readable. We read from left to right.
</blockquote></details>

Done.

**Review by osalyk — COMMENTED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195#-:-MhN3PQz9CJqlvCxjbar:bnfp4nl)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

**Review by ldorau — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195#-:-MhNFQjjEYBRru02rCNt:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

**Review by ldorau — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: all files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

___
*[tools/perf/lib/Figure.py, line 158 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDe5Kz5TOg_cyNzLrj:-MhNFTPn3gkhPAL1Apdt:b-cwdch7) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L158)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

No. Because it may give you an impression everything is fine whereas one of them may be missing. But when you see a text not in plain English with underscores instead of spaces you win two times:
1. you know something went wrong
2. you know what the argument was which has caused an issue
</blockquote></details>

@lukaszstolarczuk ping
___
*[tools/perf/lib/Figure.py, line 159 at r1](https://reviewable.io/reviews/pmem/rpma/1195#-MhDeQhR3SOrYfoqUjLV:-MhNFWK79dqi-ILdqH5P:bp3bf0v) ([raw file](https://github.com/pmem/rpma/blob/221fa6d17807c7978e9983c49f41cc043ac46a44/tools/perf/lib/Figure.py#L159)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

@lukaszstolarczuk  ping

**Review by janekmi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195)* status: all files reviewed, 2 unresolved discussions (waiting on @lukaszstolarczuk)

*[a discussion](https://reviewable.io/reviews/pmem/rpma/1195#-Mhlb9PR6EGTgTSNsig4:-Mhlb9PR6EGTgTSNsig5:bgeffji) (no related file):*
@lukaszstolarczuk is on holiday. I hope he will take a look when he will come back. In case he would like to change something I will submit another PR. In the meantime, I think this change is ready to be merged.

---

**Review by lukaszstolarczuk — COMMENTED** [review]
:+1: 

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/rpma/1195#-:-MiVQ6Mf3cj6igiy-XRt:b-8fx2u7)* status: :shipit: complete! all files reviewed, all discussions resolved
